File: 061/rotate-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None


class Solution:
    def rotateRight(self, head, k):
        """
        :type head: ListNode
        :type k: int
        :rtype: ListNode
        """
        if head is None or head.next is None:
            return head

        node = head
        length = 1
        while node.next is not None:
            length += 1
            node = node.next

        node.next = head
        k %= length

        for i in range(length-k):
            node = node.next

        head = node.next
        node.next = None
        return head



class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

    def __eq__(self, other):
        return self.val == other.val and self.next == other.next

File: 061/rotate-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func rotateRight(head *ListNode, k int) *ListNode {
	if head == nil || head.Next == nil {
		return head
	}

	node := head
	size := 1
	for node.Next != nil {
		size += 1
		node = node.Next
	}

	node.Next = head
	k %= size

	for i := 0; i < size-k; i++ {
		node = node.Next
	}
	head = node.Next
	node.Next = nil
	return head
}